spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/observation/TimerObservationHandlerMeterRegistryCustomizer.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2012-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.observation;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.observation.TimerObservationHandler;
+import io.micrometer.observation.ObservationRegistry;
+
+import org.springframework.boot.actuate.autoconfigure.metrics.MeterRegistryCustomizer;
+
+/**
+ * Registers the {@link TimerObservationHandler} with a {@link ObservationRegistry}.
+ *
+ * @author Moritz Halbritter
+ */
+class TimerObservationHandlerMeterRegistryCustomizer implements MeterRegistryCustomizer<MeterRegistry> {
```

Comment:
Could this be an `ObservationRegistryCustomizer` that has the `MeterRegistry` passed into its constructor and then configures the `ObservationRegistry` when it's called?

Reply:
Yup, it's more obvious your way 'round.